qutip/visualization.py

```diff
@@ -582,14 +597,22 @@ def matrix_histogram_complex(M, xlabels=None, ylabels=None,
         ax.set_title(title)
 
     # x axis
-    ax.axes.w_xaxis.set_major_locator(plt.IndexLocator(1, -0.5))
+    xtics = -0.5 + np.arange(M.shape[0])
+    ax.axes.w_xaxis.set_major_locator(plt.FixedLocator(xtics))
     if xlabels:
+        nxlabels = len(xlabels)
+        if nxlabels != len(xtics):
+            raise ValueError(f"got {nxlabels} xlabels but needed {len(xtics)}")
         ax.set_xticklabels(xlabels)
     ax.tick_params(axis='x', labelsize=12)
 
     # y axis
-    ax.axes.w_yaxis.set_major_locator(plt.IndexLocator(1, -0.5))
+    ytics = -0.5 + np.arange(M.shape[1])
+    ax.axes.w_yaxis.set_major_locator(plt.FixedLocator(ytics))
     if ylabels:
+        nylabels = len(ylabels)
+        if nylabels != len(ytics):
+            raise ValueError(f"got {nylabels} ylabels but needed {len(ytics)}")
```

Comment:
These seem like code changes, not docstring formatting. Maybe they should get their own PR? Or I can review them here if you tell me what problem they are addressing.

Reply:
Oh yeah, I forgot I did that too - I did that bit first (and rebased after).  It (and the other stuff in that same file) fixes a warning issued by matplotlib when calling `hinton` and another couple of functions caused by mixing an automatic tic-locator with hard-coded tic labels.  See commit c1147ce for this change only, and its commit message for a little more detail.

Reply:
I reviewed c1147ce an it looks good.